exp/api/v1beta2/awsmachinepool_webhook_test.go

```diff
@@ -174,6 +174,41 @@ func TestAWSMachinePoolValidateCreate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "invalid, MarketType set to MarketTypeCapacityBlock and spotMarketOptions are specified",
+			pool: &AWSMachinePool{
+				Spec: AWSMachinePoolSpec{
+					AWSLaunchTemplate: AWSLaunchTemplate{
+						MarketType:        infrav1.MarketTypeCapacityBlock,
+						SpotMarketOptions: &infrav1.SpotMarketOptions{},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid, MarketType set to MarketTypeOnDemand and spotMarketOptions are specified",
+			pool: &AWSMachinePool{
+				Spec: AWSMachinePoolSpec{
+					AWSLaunchTemplate: AWSLaunchTemplate{
+						MarketType:        infrav1.MarketTypeOnDemand,
+						SpotMarketOptions: &infrav1.SpotMarketOptions{},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid MarketType set to MarketTypeCapacityBlock is specified",
+			pool: &AWSMachinePool{
+				Spec: AWSMachinePoolSpec{
+					AWSLaunchTemplate: AWSLaunchTemplate{
+						MarketType: infrav1.MarketTypeCapacityBlock,
+					},
+				},
+			},
+			wantErr: false,
```

Comment:
Same here, this is an error

Reply:
added the webhook and added one more test cases.